### PR TITLE
Rename `ManagerInterfaceFactoryInterface`

### LINKED ISSUE #520: Rename `ManagerInterfaceFactoryInterface` to `ManagerImplementationFactoryInterface` 🤣
## What

- Rename `ManagerInterfaceFactoryInterface` to `ManagerImplementationFactoryInterface`.
- ~Make `identifiers`/`instantiate` protected~
- ~Friend `ManagerFactory`~

## Why (you only just made it)

When adding `ManagerFactory` (See #507), it caused us to do a double-take on which namespace `ManagerInterfaceFactoryInterface` should be in, as it _is_ host centric, but a host should not use one directly (just make one of the derived classes). We also found it hard to articulate the difference in the two factories in a host-friendly way (given a ~hose~ ~horse~ host is not supposed to know about `ManagerInterface`).

This hopefully makes it more obvious, and hides the methods that a host shouldn't be tempted into calling.

### PR DESCRIPTION
**Pull request by foundrytom**
Following on from #535, renames to `ManagerImplementationFactoryInterface`.

~On hold, pending #535.~
Closes #520.

### CODE REVIEW

File: python/openassetio/pluginSystem/ManagerPlugin.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def interface(cls):
         to be bound to the Host-facing @ref openassetio.hostApi.Manager.
 
         Generally this is only directly called by the @ref
-        openassetio.pluginSystem.PluginSystemManagerInterfaceFactory. It
+        openassetio.pluginSystem.PluginSystemManagerImplementationFactory. It
```
Comment (feltech) [inline python/openassetio/pluginSystem/ManagerPlugin.py:80]: [consider][minor] 72 char

File: python/openassetio/pluginSystem/PluginSystemManagerImplementationFactory.py [hunk 2]
```diff
@@ -14,22 +14,22 @@
 #   limitations under the License.
 #
 """
-@namespace openassetio.pluginSystem.PluginSystemManagerInterfaceFactory
-A single-class module, providing the PluginSystemManagerInterfaceFactory
+@namespace openassetio.pluginSystem.PluginSystemManagerImplementationFactory
+A single-class module, providing the PluginSystemManagerImplementationFactory
```
Comment (feltech) [inline python/openassetio/pluginSystem/PluginSystemManagerImplementationFactory.py:18]: [consider][minor] 72 char

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -37,8 +43,8 @@ v1.0.0-alpha.X
   [#457](https://github.com/OpenAssetIO/OpenAssetIO/issues/457)
 
 - Renamed both `ManagerFactoryInterface` as well as the subclass
-  `PluginSystemManagerFactory` to `ManagerInterfaceFactoryInterface`
-  and `PluginSystemManagerInterfaceFactory`, respectively, to avoid
+  `PluginSystemManagerFactory` to `ManagerImplementationFactoryInterface`
```
Comment (feltech) [inline RELEASE_NOTES.md:46]: [consider][minor] 72 char

File: RELEASE_NOTES.md [hunk 4]
```diff
@@ -145,7 +151,7 @@ v1.0.0-alpha.X
 
 - Migrated the following classes to C++: `Context`, `Host`,
   `HostInterface`, `HostSession` and `LoggerInterface`,
-  `ManagerInterfaceFactoryInterface`. Debug and audit functionality is
+  `ManagerImplementationFactoryInterface`. Debug and audit functionality is
```
Comment (feltech) [inline RELEASE_NOTES.md:154]: [consider][minor] 72 char

**Review by feltech — APPROVED** [review]

File: RELEASE_NOTES.md (hunk 4 above)
Comment (foundrytom) [reply RELEASE_NOTES.md]: Fixed [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/536/commits/d96ad0627dfa3c6abd59a2efce451da2d3305c04).

File: RELEASE_NOTES.md (hunk 3 above)
Comment (foundrytom) [reply RELEASE_NOTES.md:46]: I left this one, as it was really hard to read, and it only seems to be over by 1

File: python/openassetio/pluginSystem/PluginSystemManagerImplementationFactory.py (hunk 2 above)
Comment (foundrytom) [reply python/openassetio/pluginSystem/PluginSystemManagerImplementationFactory.py]: Fixed [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/536/commits/d96ad0627dfa3c6abd59a2efce451da2d3305c04).

File: python/openassetio/pluginSystem/ManagerPlugin.py (hunk 1 above)
Comment (foundrytom) [reply python/openassetio/pluginSystem/ManagerPlugin.py]: Fixed [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/536/commits/d96ad0627dfa3c6abd59a2efce451da2d3305c04).

**Review by feltech — APPROVED** [review]
LGTM squash and merge when ready

**Pull request merged**